File: frontend/src/types/history.ts
export interface ScanHistoryItem {
  id: string;
  farmId: string;
  farmName: string;
  crop: string;
  condition: string;
  confidence: number;
  status: 'Needs Attention' | 'Healthy-looking' | 'Uncertain';
  scanDate: string;
  thumbnailUrl: string;
  healthScoreContribution: number;
}

export interface FarmNotification {
  id: string;
  title: string;
  message: string;
  category: 'scanner' | 'weather' | 'irrigation' | 'system';
  isRead: boolean;
  timestamp: string;
  linkUrl?: string;
}

File: backend/app/schemas/history.py
from typing import Optional, Literal
from pydantic import BaseModel, Field, ConfigDict

ScanStatusType = Literal["Needs Attention", "Healthy-looking", "Uncertain"]


class ScanHistoryItemResponse(BaseModel):
    id: str
    farmId: str = Field(..., serialization_alias="farmId")
    farmName: str = Field(..., serialization_alias="farmName")
    crop: str
    condition: str
    confidence: float
    status: ScanStatusType
    scanDate: str = Field(..., serialization_alias="scanDate")
    thumbnailUrl: str = Field(..., serialization_alias="thumbnailUrl")
    healthScoreContribution: int = Field(..., serialization_alias="healthScoreContribution")

    model_config = ConfigDict(populate_by_name=True)
